app.js

```diff
@@ -127,4 +142,21 @@ document.getElementById('contact-form').addEventListener('submit', (e) => {
       .getElementById('email_error_lowercase')
       .classList.remove('no-visible');
   }
-});
\ No newline at end of file
+});
+
+/*
+store form data
+*/
+window.addEventListener('load', () => {
+  formData = JSON.parse(localStorage.getItem('formData'));
+  if (localStorage.getItem('formData') != null) {
+    document.querySelectorAll('[required]').forEach((formElement) => {
+      formElement.value = formData[formElement.id];
+    });
+  }
+});
+
+document.getElementById('contact-form').addEventListener('keyup', (e) => {
+  formData[e.target.id] = e.target.value;
+  localStorage.setItem('formData', JSON.stringify(formData));
+});
```

Comment:
Good work creating an event listener, but the local storage is not working properly. 🔴 

- [ ] Kindly, when a user navigates to the webpage and fill in the form, without submitting it, the data on the inputs should be saved on a formData object {}.
- [ ] When a user it's refreshing the webpage the data on the forms should be preserved on the local storage. You could use a function like this 💯 

window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};